sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp

```diff
@@ -62,10 +62,10 @@ static const std::vector<std::string> availableControls = {
 static const std::string skEepromName = "custom";
 CameraFxTof1::CameraFxTof1(
     std::shared_ptr<aditof::DepthSensorInterface> depthSensor,
-    std::shared_ptr<aditof::StorageInterface> eeprom,
-    std::shared_ptr<aditof::TemperatureSensorInterface> temperatureSensor)
-    : m_depthSensor(depthSensor), m_eeprom(eeprom),
-      m_temperatureSensor(temperatureSensor), 
+    std::vector<std::shared_ptr<aditof::StorageInterface>> &eeprom,
+    std::vector<std::shared_ptr<aditof::TemperatureSensorInterface>> &tSensor)
+    : m_depthSensor(depthSensor), m_eeprom(eeprom.first()),
```

Comment:
What happens if the vector is empty?
Are we sure that this will never happen?

Reply:
Actually, I forgot to rework the FXTOF1 camera. This won't even compile.